lib/xref.h

```diff
@@ -139,8 +144,10 @@ extern const struct xref * const __stop_xref_array[1] DSO_LOCAL;
  */
 #define XREF_SETUP()                                                           \
 	static const struct xref _dummy_xref = {                               \
-			.file = __FILE__, .line = __LINE__, .func = "dummy",   \
 			.type = XREFT_NONE,                                    \
+			.line = __LINE__,                                      \
+			.file = __FILE__,                                      \
+			.func = "dummy",                                       \
```

Comment:
C++ doesn't understand designated initializers until C++20, I have a separate patch set to fix this or you can fix it here, your call

Reply:
Hm, I thought both clang and gcc had accepted designated initializers in C++ mode for a long time, but I'm very much not a C++ developer...  if you've already fixed it, I'm happy to just drop it from this PR.

Reply:
Amended to remove this change.